source/kernel/src/registry/mod.rs

```diff
@@ -167,6 +171,13 @@ pub struct Envelope<P> {
     request_id: RequestResponseId,
 }
 
+pub struct OpenEnvelope<P> {
```

Comment:
a doc comment explaining this might be nice?
also, should probably be
```suggestion
#[must_use]
pub struct OpenEnvelope<P> {
```

Reply:
Gunna defer on this because I want to revisit this and move the whole API over to this to get rid of the old bad one.